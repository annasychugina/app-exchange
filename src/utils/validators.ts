import {Currency, CurrentBalanceConfig} from '../types/types';

const errorMessage = 'You do not have enough money in your account';
const errorRequired = 'No values for exchange';
export const checkBalance = ({
  currencyFrom,
  valueFrom,
  userBalance,
}: {
  currencyFrom: Currency;
  valueFrom: number | null;
  userBalance: CurrentBalanceConfig;
}): {isValid: boolean; errorMessage?: string} => {
  if (!valueFrom) {
    return {isValid: false, errorMessage: errorRequired};
  }
  if (valueFrom > userBalance[currencyFrom]) {
    return {isValid: false, errorMessage};
  }

  return {isValid: true};
};
