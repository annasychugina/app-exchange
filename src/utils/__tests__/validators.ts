import {checkBalance} from '../validators';

test('validators: checkBalance', () => {
  expect(
    checkBalance({
      currencyFrom: 'RUB',
      valueFrom: null,
      userBalance: {
        RUB: 123,
        USD: 12,
      },
    }),
  ).toMatchSnapshot();

  expect(
    checkBalance({
      currencyFrom: 'RUB',
      valueFrom: 124,
      userBalance: {
        RUB: 123,
        USD: 12,
      },
    }),
  ).toMatchSnapshot();

  expect(
    checkBalance({
      currencyFrom: 'RUB',
      valueFrom: 123,
      userBalance: {
        RUB: 123,
        USD: 12,
      },
    }),
  ).toMatchSnapshot();
});
