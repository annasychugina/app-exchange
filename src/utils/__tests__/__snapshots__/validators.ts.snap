// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validators: checkBalance 1`] = `
Object {
  "errorMessage": "No values for exchange",
  "isValid": false,
}
`;

exports[`validators: checkBalance 2`] = `
Object {
  "errorMessage": "You do not have enough money in your account",
  "isValid": false,
}
`;

exports[`validators: checkBalance 3`] = `
Object {
  "isValid": true,
}
`;
